components/tidb_query_expr/src/impl_time.rs

```diff
@@ -696,6 +730,75 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_add_sub_datetime_and_duration() {
```

Comment:
Split `test_add_sub_datetime_and_duration` into two tests will make tests more clearly, what do you think?

Reply:
Yes, maybe the logic between `add_datetime_and_duration` and `sub_datetime_and_duration` are different...

done